### DISCUSSION TITLE
Not getting the session of new driver in other test_aa.....

### DISCUSSION THREAD

**Discussion by Jeremy-QA** [seleniumbase/SeleniumBase#2104] [General]
Hello, how to get the session browser of self.get_new_driver() to run in other test_aa. Currently i am using --reuse-session and running smoothly. But when I declare self.get_new_driver() in my page object it will not continue executing

**Comment by mdmintz** [comment]
Hi @Jeremy-QA, `--reuse-session` is for having all tests reuse the original browser session that was spun up. If you use `self.get_new_driver()` to open up a new browser, then it won't be part of the reused browser session chain. If something were to happen to the original browser, (eg. it was closed manually), then SeleniumBase will try to continue reusing a session on a new browser.

**Comment by Jeremy-QA** [comment]
Hi @mdmintz thank you for your response. I have so much fun learning your framework. Can I ask how to achieve or the best way to continue reusing a session on self.get_new_driver().

Here is my code. In My case the execution of will stop at test_b

ClassA (BaseCase)

def open_original_window(self):
      self.open("example url")

def open_new_window(self,invitation_link ):
        self.get_new_driver()
        self.maximize_window()
        self.open(invitation_link)
        
  
        
ClassB (BaseCase)

def continue_testing(self):
        self.click(self.element)
        
        
 Class MainTest(ClassA, ClassB)
        
@pytest.mark.run(order=1)
def test_a(self):
          self.open_original_window()
             
@pytest.mark.run(order=2)
def test_b(self):
           self.open_new_window("example url")
              
@pytest.mark.run(order=3)
def test_c(self):
            self. continue_testing()

**Comment by mdmintz** [comment]
Here are a few examples to explain `--rs` mode:

Example 1:

`test_1` opens `Page A`. `test_2` starts on `Page A`, but then immediately opens a new driver window and goes to `Page B`. When `test_3` starts, it sees 2 open windows, so it picks the first one to reuse (`Page A`, because the first window is the default `shared_driver`) and closes the others (`Page B`). Then it navigates to `Page C` on the window that had `Page A`.

```python
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__, "--rs")

class MyClass(BaseCase):
    def test_1(self):
        self.open("data:text/html,<h1>Page A</h1>")
        self.sleep(2)

    def test_2(self):
        self.get_new_driver()
        self.open("data:text/html,<h1>Page B</h1>")
        self.sleep(2)

    def test_3(self):
        self.open("data:text/html,<h1>Page C</h1>")
        self.sleep(2)
```

Example 2: (Add an import `from seleniumbase import config as sb_config`, and adds `sb_config.shared_driver = self.driver` after calling `self.get_new_driver()`):

`test_1` opens `Page A`. `test_2` starts on `Page A`, but then immediately opens a new driver window, changes the `shared_driver`, and goes to `Page B`. When `test_3` starts, it reuses the shared driver (`Page B`) and closes the others (`Page A`). Then it navigates to `Page C` on the window that had `Page B`.

```python
from seleniumbase import BaseCase
from seleniumbase import config as sb_config
BaseCase.main(__name__, __file__, "--rs")

class MyClass(BaseCase):
    def test_1(self):
        self.open("data:text/html,<h1>Page A</h1>")
        self.sleep(2)

    def test_2(self):
        self.get_new_driver()
        sb_config.shared_driver = self.driver
        self.open("data:text/html,<h1>Page B</h1>")
        self.sleep(2)

    def test_3(self):
        self.open("data:text/html,<h1>Page C</h1>")
        self.sleep(2)
```

To summarize: Use `sb_config.shared_driver = self.driver` after using `self.get_new_driver()` if you want that new driver to be reused by future tests. Otherwise, new tests will try to reuse the first browser session that was opened.

**Comment by Jeremy-QA** [comment]
it works 👍 💯  thank you so @mdmintz

**Reply by Jeremy-QA** [reply]
hi @mdmintz  is there a way to prevent the previous driver from auto closing the driver based to the next test on my design of code that we discussed? thank you

**Reply by mdmintz** [reply]
Set `sb_config.shared_driver` to the driver you want to keep / reuse.

**Reply by Jeremy-QA** [reply]
hello @mdmintz this is what i need, when I run pytest python_file.py --rs. test_1 will open facebook and open new driver to open youtube.

Upon executing, when test 2 is running. The facebook tab will close which is i need to stay open

           from seleniumbase import BaseCase
           from seleniumbase import config as sb_config
           BaseCase.main(__name__, __file__, "--rs")


          class MyClass(BaseCase):
                def test_1(self):
                     self.open("https://www.facebook.com")
                     self.sleep(2)
                     self.get_new_driver()
                     sb_config.shared_driver = self.driver
                     self.open("https://www.youtube.com")

            def test_2(self):
                  self.refresh_page()

**Reply by mdmintz** [reply]
Then remove your `sb_config.shared_driver = self.driver` line, which caused your youtube driver to stay open instead.

**Reply by Jeremy-QA** [reply]
can you gave me an example of multiple drivers in page object that can be able switch back to the chosen driver and all the drivers will remain opened?. thank you so much for your help :)

**Reply by mdmintz** [reply]
Only one driver gets to be passed to other tests in reuse-session mode. By default, it's the first driver. If you want that reused driver to be a different driver, use `sb_config.shared_driver = self.driver` after you've obtained it in a test. The next test that runs will reuse that one.

**Reply by Jeremy-QA** [reply]
thank you so much @mdmintz . Very informative information.
